packages/common/src/select/select.js

```diff
@@ -5,6 +5,36 @@ import PropTypes from 'prop-types';
 import clsx from 'clsx';
 import useSelect from '../use-select/use-select';
 
+const deepEqual = (a, b, depth = 0) => {
```

Comment:
Could we provide some unit tests for this function and extract it to a separate file?

Reply:
Sure